lib/templates.js

```diff
@@ -173,31 +177,40 @@ const createAPIResourceTemplate = (serviceName, serviceDefinition, appConfig, pa
 
             ...ordExtensions,
         };
-
-        apiResources.push(obj);
+        // visibility can also be set as an ORD extension; in this case, it overwrites the default visibility
+        if (obj.visibility === RESOURCE_VISIBILITY.public) apiResources.push(obj);
     });
 
-    if (apiResources.length > 0) return apiResources;
+    return apiResources;
 };
 
-const createEventResourceTemplate = (serviceName, eventDefinition, appConfig, packageIds) => {
-    const ordExtensions = readORDExtensions(eventDefinition);
-    return {
+/**
+ * This is a template function to create Event Resource object for Event Resource Array.
+ * There can be only one event resource per service because all events are using the same protocol, they are always Cloud Events.
+ *
+ * @param {string} serviceName The name of the service.
+ * @param {object} serviceDefinition The definition of the service
+ * @returns {Array} An single-item array of objects for the Event Resources.
+ */
```

Comment:
@zongqichen : I think that the line 189 is needed because it's not obvious from the code that there can be only one event resource per service(because all the events are using the same protocol). This information seems useful for me.
@Fannon @swennemers : Do you agree to keep this doc comment?

Reply:
@aramovic79 , @zongqichen - please align on the code style around comments.